src/provider/controller/NotificationCenter.ts

```diff
@@ -1,145 +1,189 @@
+                // if (!contextMenuIsShowing && this.visible) {
+                if (this.visible) {
+                    this._store.dispatch(toggleCenterWindowVisibility(false));
                 }
-            );
-        });
+            });
+        }
+        fin.System.addListener('monitor-info-changed', ((event: MonitorEvent<string, string>) => {
+            this.sizeToFit();
+        }));
     }
 
     /**
-     * @function showWindow Shows the Window with Fade()
-     * @returns void
+     * Toggle window visibility.
+     * @param forceVisibility Force the window to be shown or hidden. True = show, false = hide.
      */
-    public showWindow(): void {
-        this.fade(true, 450);
-        this.windowInfo.getWindow().resizeBy(1, 0, 'top-left');
-        this.windowInfo.getWindow().resizeBy(-1, 0, 'top-left');
-        this.windowInfo.setShowing(true);
+    public async toggleWindow(forceVisibility?: boolean): Promise<void> {
+        const visible = forceVisibility || this.visible;
+        if (visible) {
+            return this.showWindow();
+        } else {
+            return this.hideWindow();
+        }
     }
 
     /**
-     * @function hideWindow Hides the window with Fade()
-     * @returns void
+     * Show the window.
      */
-    public hideWindow(): void {
-        this.fade(false, 450);
-        this.windowInfo.setShowing(false);
+    public async showWindow(): Promise<void> {
+        const {window} = await this._webWindow;
+        window.bringToFront();
+        await window.show();
+        await this.animateIn();
+        window.setAsForeground();
     }
 
     /**
-     * @function toggleWindow Hides or shows the center.
-     * @returns void
+     * Hide the window.
+     * @param force Force the window to hide without animating out.
      */
-    public toggleWindow(): void {
-        if (this.windowInfo.getShowingStatus()) {
-            this.hideWindow();
-        } else {
-            this.showWindow();
-        }
+    public async hideWindow(force?: boolean): Promise<void> {
+        const duration = force ? 0 : 300;
+        await this.animateOut(duration);
     }
 
     /**
-     * @function fade Fades the window in or out
-     * @param fadeOut
-     * @param timeout
+     * @function sizeToFit Sets the window dimensions in shape of a side bar
      */
-    public fade(fadeOut: boolean, timeout: number): void {
-        if (fadeOut) {
-            this.windowInfo.getWindow().show();
-        }
-
-        this.windowInfo.getWindow().animate({opacity: {opacity: fadeOut ? 1 : 0, duration: timeout}}, {interrupt: false}, () => {
-            !fadeOut ? this.windowInfo.getWindow().hide() : this.windowInfo.getWindow().focus();
+    public async sizeToFit(): Promise<void> {
+        const {window} = await this._webWindow;
+        await this.hideWindow(true);
+        const monitorInfo = await fin.System.getMonitorInfo();
+        const idealWidth = 388;
+        window.setBounds({
```

Comment:
wondering if a `return` here might make sense in case we would ever want to `await` it?